### PR TITLE
[15.0][MIG] sale: migration script

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Supersedes https://github.com/OCA/OpenUpgrade/pull/3342.

### CODE REVIEW

**Review by remi-filament — APPROVED** [review]
tested OK for our database migration, thank you @MiquelRForgeFlow

File: openupgrade_scripts/scripts/sale/15.0.1.2/pre-migration.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.convert_field_to_html(
+        env.cr, "sale_order", "note", "note", verbose=False
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/sale/15.0.1.2/pre-migration.py]: Copy the column on pre to preserve the old contents, and convert it in post

File: openupgrade_scripts/scripts/sale/15.0.1.2/end-migration.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    all_sale_order_lines = env["sale.order.line"].search(
+        [("product_packaging_id", "!=", False)]
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/sale/15.0.1.2/end-migration.py]: This should be restricted to quotations, as we don't want past orders to be modified on the migration. It's legit to do it for keeping a consistent state on pending ones:

```suggestion
        [("product_packaging_id", "!=", False), ('state', 'in', ['draft', 'sent'])]
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**